os/board/rtl8721csm/src/component/common/bluetooth/realtek/sdk/example/ble_peripheral/ble_tizenrt_server.c

```diff
@@ -740,73 +622,33 @@ trble_result_e rtw_ble_server_delete_bonded_device_all(void)
 
 int rtw_ble_server_set_adv_interval(unsigned int interval)
 {
-    T_GAP_DEV_STATE new_state;
-    le_get_gap_param(GAP_PARAM_DEV_STATE , &new_state);
+    rtw_ble_server_adv_into_idle();
 
-    switch (new_state.gap_adv_state)
-    {
-    case GAP_ADV_STATE_ADVERTISING:
-        {
-            debug_print("\r\n[%s] ADV STATE : ADVERTISING", __FUNCTION__);
-            if(ble_tizenrt_server_send_msg(BLE_TIZENRT_MSG_STOP_ADV, NULL) == false)
-            {
-                debug_print("\r\n[%s] msg send fail", __FUNCTION__);
-                return TRBLE_FAIL;
-            }
-        }
-        break;
-    case GAP_ADV_STATE_START:
-        {
-            do {   
-                debug_print("\r\n[%s] ADV STATE : START", __FUNCTION__);
-                os_delay(100);
-                le_get_gap_param(GAP_PARAM_DEV_STATE , &new_state);
-            } while(new_state.gap_adv_state == GAP_ADV_STATE_START);
-
-            if(new_state.gap_adv_state == GAP_ADV_STATE_ADVERTISING)
-            {
-                debug_print("\r\n[%s] ADV STATE : ADVERTISING", __FUNCTION__);
-                if(ble_tizenrt_server_send_msg(BLE_TIZENRT_MSG_STOP_ADV, NULL) == false)
-                {
-                    debug_print("\r\n[%s] msg send fail", __FUNCTION__);
-                    return TRBLE_FAIL;
-                }
-            }
-        }
-        break;
-    case GAP_ADV_STATE_STOP:
-        {
-            do {   
-                debug_print("\r\n[%s] ADV STATE : STOPPING", __FUNCTION__);
-                os_delay(100);
-                le_get_gap_param(GAP_PARAM_DEV_STATE , &new_state);
-            } while(new_state.gap_adv_state == GAP_ADV_STATE_STOP);
+    uint16_t adv_int = interval;
+    le_adv_set_param(GAP_PARAM_ADV_INTERVAL_MIN, sizeof(adv_int), &adv_int);
+    le_adv_set_param(GAP_PARAM_ADV_INTERVAL_MAX, sizeof(adv_int), &adv_int);
+    return TRBLE_SUCCESS;
+}
 
-            if(new_state.gap_adv_state == GAP_ADV_STATE_ADVERTISING)
-            {
-                debug_print("\r\n[%s] ADV STATE : ADVERTISING", __FUNCTION__);
-                if(ble_tizenrt_server_send_msg(BLE_TIZENRT_MSG_STOP_ADV, NULL) == false)
-                {
-                    debug_print("\r\n[%s] msg send fail", __FUNCTION__);
-                    return TRBLE_FAIL;
-                }
-            }            
-        }
-        break;
-    default:
-        break;
+int rtw_ble_server_set_adv_type(trble_adv_type_e type, void *param)
+{
+    if(type == TRBLE_ADV_TYPE_DIRECT && param == NULL)
```

Comment:
Is this `&&` intended?
I mean, is it okay the `type` is not `TRBLE_ADV_TYPE_DIRECT` but `param` is NULL?

Reply:
Yes. It is intended. But I have optimized this API.